pkg/controller.v1/control/podgroup_control.go

```diff
@@ -125,23 +125,28 @@ var _ PodGroupControlInterface = &VolcanoControl{}
 
 // SchedulerPluginsControl is the  implementation of PodGroupControlInterface with scheduler-plugins.
 type SchedulerPluginsControl struct {
-	Client client.Client
+	Client        client.Client
+	SchedulerName string
 }
 
 func (s *SchedulerPluginsControl) DecoratePodTemplateSpec(pts *corev1.PodTemplateSpec, job metav1.Object, _ string) {
+	if len(pts.Spec.SchedulerName) == 0 {
+		pts.Spec.SchedulerName = s.GetSchedulerName()
+	}
+
 	if pts.Labels == nil {
 		pts.Labels = make(map[string]string)
 	}
 	pts.Labels[schedulerpluginsv1alpha1.PodGroupLabel] = job.GetName()
 }
 
 func (s *SchedulerPluginsControl) GetSchedulerName() string {
-	return ""
+	return s.SchedulerName
 }
 
 // NewSchedulerPluginsControl returns a SchedulerPluginsControl
-func NewSchedulerPluginsControl(c client.Client) PodGroupControlInterface {
-	return &SchedulerPluginsControl{Client: c}
+func NewSchedulerPluginsControl(c client.Client, schedulerName string) PodGroupControlInterface {
```

Comment:
Not about this PR, but I realized that some functions below are unnecessary, like `UpdatePodGroup`, and `CreatePodGroup`. They are casting unnecessarily: the controller-runtime client can already handle the casting via reflection. And I think `GetPodGroup` and `DeletePodGroup` could be refactored so that we don't need the casting either. Maybe even use generics.